File: server/router/kubernetes/roleBinding.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type RoleBindingRouter struct{}

func (r *RoleBindingRouter) InitRoleBindingRouter(Router *gin.RouterGroup) {
	roleBindingRouter := Router.Group("roleBinding").Use(middleware.OperationRecord())
	roleBindingRouterWithoutRecord := Router.Group("roleBinding")
	roleBindingApi := v1.ApiGroupApp.KubernetesApiGroup.RoleBindingApi
	{
		roleBindingRouter.POST("getRoleBindingRaw", roleBindingApi.GetRoleBindingRaw)       // 获取oleBinding编排
		roleBindingRouter.POST("getRoleBindingDetail", roleBindingApi.GetRoleBindingDetail) // 获取roleBinding详情
		roleBindingRouter.POST("deleteRoleBinding", roleBindingApi.DeleteRoleBinding)       // 删除oleBinding
	}
	{
		roleBindingRouterWithoutRecord.POST("getRoleBindingList", roleBindingApi.GetRoleBindingList) // 获取所有roleBinding
	}
}

File: web/src/api/kubernetes/roleBinding.js
import service from '@/utils/request'

// 获取roleBinding list
export const getRoleBindingList = (data) => {
  return service({
    url: '/roleBinding/getRoleBindingList',
    method: 'post',
    data
  })
}

// 获取roleBinding编排
export const getRoleBindingRaw = (data) => {
  return service({
    url: '/roleBinding/getRoleBindingRaw',
    method: 'post',
    data
  })
}

// 获取roleBinding详情
export const getRoleBindingDetail = (data) => {
  return service({
    url: '/roleBinding/getRoleBindingDetail',
    method: 'post',
    data
  })
}

// 删除roleBinding
export const deleteRoleBinding = (data) => {
  return service({
    url: '/roleBinding/deleteRoleBinding',
    method: 'post',
    data
  })
}
